src/Symfony/Component/Notifier/Bridge/Mattermost/MattermostTransport.php

```diff
@@ -56,14 +56,15 @@ protected function doSend(MessageInterface $message): void
             throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" (instance of "%s" given).', __CLASS__, ChatMessage::class, get_debug_type($message)));
         }
 
-        $endpoint = sprintf('https://%s/api/v4/posts', $this->getEndpoint());
-
         $options = ($opts = $message->getOptions()) ? $opts->toArray() : [];
         $options['message'] = $message->getSubject();
 
         if (!isset($options['channel_id'])) {
             $options['channel_id'] = $message->getRecipientId() ?: $this->channel;
         }
+
+        $endpoint = sprintf('https://%s/api/v4/posts', $this->host);
```

Comment:
I moved it don, but the important part is to use `$this->host` instead of `$this->getEndpoint`